src/Queue/AWSSQS/SqsQueueManager.php

```diff
@@ -209,11 +216,21 @@ public function pushDelayed(JobInterface $job, int $delayInSeconds): void
     {
         $prefixedQueueName = $this->getPrefixedQueueName($job->getJobDefinition()->getQueueName());
 
-        $parameters = [
-            self::DELAY_SECONDS => $delayInSeconds,
-        ];
+        if ($delayInSeconds > self::MAX_DELAY_SECONDS) {
+            $executionPlannedAt = $this->dateTimeImmutableFactory->getNow()->modify(
+                sprintf('+ %d seconds', $delayInSeconds),
+            );
+            $this->logger->info(
+                'Requested delays is greater than SQS limit. Job execution has been planned and will be requeued until then.',
```

Comment:
type (probably mine :grin:)
```suggestion
                'Requested delay is greater than SQS limit. Job execution has been planned and will be requeued until then.',
```